### PR TITLE
CLN: Remove redundant index test from tests/base/test_ops.py

### PR DESCRIPTION
**Pull request by SaturnFromTitan**
part of #23877
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

Boolean indexing is already thoroughly tested in `test_series_mask_boolean` in `tests/indexes/test_na_indexing.py` - [see here](https://github.com/pandas-dev/pandas/blob/master/pandas/tests/indexing/test_na_indexing.py#L7-L61)

### CODE REVIEW

**Review by WillAyd — CHANGES_REQUESTED** [review]
On board with cleaning up, but are we not losing any coverage here? Maybe overlooking, but one that sticks out is indexing with an Index object which appears here but I think not in the test you linked to replace this?

**Comment by SaturnFromTitan** [comment]
Good point @WillAyd, I'll check that later. I was assuming that if we lose coverage, then I'd get a codecov warning here in the PR. Isn't there a bot for these sort of things?

**Comment by WillAyd** [comment]
We do have codecov but it’s not generally very insightful (at least for us)

File: pandas/tests/indexing/test_na_indexing.py [hunk 1]
```diff
@@ -22,42 +25,44 @@
...
-    mask = mask[:0]
-    ser = ser.iloc[:0]
-    expected = ser[mask.astype("bool")]
-    result = ser[mask]
-    tm.assert_equal(result, expected)
+    if indexer_class is pd.Series:
+        msg = "iLocation based boolean indexing cannot use an indexable as a mask"
+        expectation = pytest.raises(ValueError, match=msg)
+    else:
+        expectation = do_not_raise()
```
Comment (SaturnFromTitan) [inline pandas/tests/indexing/test_na_indexing.py]: inspired by https://github.com/pytest-dev/pytest/issues/1830

**Comment by SaturnFromTitan** [comment]
@WillAyd I think this is ready for another review now. I added an `indexer_class` parametrization to `test_series_mask_boolean` and refactored the test a little bit (which I mostly had to do to get it passing)

File: pandas/tests/indexing/test_na_indexing.py [hunk 2]
```diff
@@ -22,42 +23,43 @@
 @pytest.mark.parametrize(
     "mask", [[True, False, False], [True, True, True], [False, False, False]]
 )
-@pytest.mark.parametrize("box_mask", [True, False])
+@pytest.mark.parametrize("indexer_class", [list, pd.array, pd.Index, pd.Series])
```
Comment (SaturnFromTitan) [inline pandas/tests/indexing/test_na_indexing.py]: This test previously only covered `pd.array` and `pd.Series`

File: pandas/tests/indexing/test_na_indexing.py [hunk 3]
```diff
@@ -7,6 +7,7 @@
 @pytest.mark.parametrize(
     "values, dtype",
     [
+        ([], "object"),
```
Comment (SaturnFromTitan) [inline pandas/tests/indexing/test_na_indexing.py]: The empty scenario was previously covered with some redundant code in the bottom of the test. I moved it to the parametrization instead and adjusted the generic test code to handle it properly

**Comment by jreback** [comment]
looks ok, will have to look again after @WillAyd comments are addressed.

**Comment by SaturnFromTitan** [comment]
@jreback I addressed all comments of @WillAyd (added the lost test cases to the parametrization of `test_series_mask_boolean`)

**Comment by jreback** [comment]
thanks @SaturnFromTitan

**Pull request merged**